added capability overview pages; fixed redirect issues (#20328)
hashicorp/consul: website/content/docs/access/k8s.mdx

layout: docs
---

# Service mesh ingress on Kubernetes

This topic provides an overview of how Consul securely allows systems outside the service mesh to access services inside the mesh on Kubernetes. Network traffic that connects services inside the mesh to external clients or services is referred to as _north-south_ traffic.

For information about enabling intra-mesh, or _east-west_ , traffic refer to [Link network segments overview](/consul/docs/link).

## Introduction

You can define points of ingress to the service mesh using either API gateways or ingress gateways. These gateways allow external network clients to access applications and services running in a Consul datacenter.

API gateways forward requests from clients to specific destinations based on path or request protocol. Ingress gateways are Consul's legacy capability for ingress and have been deprecated in favor of API gateways.

> ### API gateways

Author: ```suggestion
## API gateways
```

> API gateways enable external network clients to access applications and services running in a Consul datacenter. Consul API gateways can also forward requests from clients to specific destinations based on path or request protocol.

Author: ```suggestion
API gateways enable external network clients to securely access applications and services running in a Consul datacenter. Consul API gateways can also forward requests from clients to specific destinations in the service mesh based on request's path or protocol.
```

Systems that access services in the mesh may be internal or external to your organizational network. This type of network traffic is commonly called north-south network traffic.

> Refer to [API gateway overview](/consul/docs/access/api-gateway) for additional information about deploying API gateways to your Consul service mesh.

Author: ```suggestion
To enable an API gateway, you must configure the gateway, its listeners, and its routes. Refer to [API gateway overview](/consul/docs/access/api-gateway) for additional information about deploying API gateways to your Consul service mesh.
```

> ### Ingress gateways

Author: ```suggestion
## Ingress gateways
```

> Ingress gateways listen for external requests and route authorized traffic to instances in the service mesh. They provide one-way traffic from external sources to services in the mesh. If you want to enable traffic from services in the mesh to external destinations, then you must also configure a _terminating gateway_.  Refer to [Terminating gateways overview](/consul/docs/register/external/terminating-gateway).
> 
> Note that terminating gateways are separate components from ingress gateways and require additional configuration and maintenance. Use API gateways to enable north-south traffic Instead of implementing ingress gateways and terminating gateways.
> 
> Refer to [Ingress gateway overview](/consul/docs/access/ingress-gateway) for additional information about deploying ingress gateways to your Consul service mesh.

Author: ```suggestion
Ingress gateways listen for external requests and route authorized traffic to instances in the service mesh. They provide one-way traffic from external sources to services in the mesh. If you want to enable traffic from services in the mesh to external destinations, then you must also configure a [terminating gateway](/consul/docs/register/external/terminating-gateway), which is a separate component that requires additional configuration and maintenance.

We recommend using API gateways to enable north-south traffic instead of implementing ingress gateways and terminating gateways.

Refer to [Ingress gateway overview](/consul/docs/access/ingress-gateway) for additional information about deploying ingress gateways to your Consul service mesh.
```

<Note>

> Ingress gateways are deprecated and will no longer be updated. Ingress gateways are fully supported in this version, but we will remove them in a future release of Consul. Use [Consul API gateways](/access/api-gateway) instead.

Author: ```suggestion
Ingress gateways are deprecated. Use [Consul API gateways](/access/api-gateway) to secure service mesh ingressinstead.
```
